### DISCUSSION TITLE
16.01.2024

### DISCUSSION THREAD

**Discussion by Roadraudi** [Roadraudi/Codespace_Repository#11] [Announcements]
Hi Leute,

in meinem Repository sind ab letzter Woche bis heute wieder neue Programme dazu gekommen. Einmal ein relativ einfaches Programm das 3 Zahlenwerte entgegen nimmt und danach auf Eingabe des Benutzers die größte oder kleinste Zahl bestimmt also ein [zahlenVergleich](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/zahlenVergleich.py). 
Zudem habe ich einmal noch mein [Schere-Stein-Papier](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Rock-Paper-Scissors.py) Spiel ein Computer hinzugefügt, der seine Schwierigkeit ändern kann. Diese Funktion funktioniert aber nur wenn der Spieler gewinnt, denn dann holt sich der Computer in der nächsten Runde die letzte Eingabe des Spielers und nimmt dann das was in der letzten Runde den Spieler geschlagen hätte oder das was nicht genommen wurde. Bei der Schwierigkeit geht man so vor das man davon ausgeht das der Spieler nicht nochmal das gleiche nimmt und somit entweder der Computer gewinnt oder Unentschieden spielt. Außerdem habe ich mal noch meine eigene [Lösung](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Schere-Stein-Papier.py) hochgeladen, die wir in der Vorlesung am 08.01.24 hatten, die sich meiner Meinung nach schon sehr von dem anderen Spiel unterscheidet. Ihr könnt ja selbst entscheiden was ihr besser findet!

So und jetzt kommen wir noch zu meinem [Passwortgenerator](https://github.com/Roadraudi/Codespace_Repository/blob/codespace-upgraded-space-trout-r4466vjqx6wgf5vvv/src/Passwortgenerator.py) den ich übers Wochenende programmiert habe. Dabei kann der Nutzer ein neues Passwort erstellen auf zwei Varianten. Entweder erstellt eine Funktion ein neues Passwort, dabei kann der Nutzer entscheiden ob nur Groß- und Kleinbuchstaben verwendet werden oder auch Zahlen sowie Sonderzeichen. Außerdem entscheidet der Nutzer wie Lange sein Passwort sein soll. Um das Umzusetzen habe ich String importiert, denn dieses Modul beinhaltet alle Buchstaben, Zahlen und Sonderzeichen als Konstanten und somit musste ich diese nicht jeweils einzeln alle in zum Beispiel einer Liste speichern. [Hier](https://stackoverflow.com/questions/7785726/importing-using-string-in-python) findet ihr mehr zu dem Modul String und was man alles cooles damit machen kann.
Bei der zweiten Variante kann der Nutzer über die Eingabe beliebig viele Buchstaben, Zeichen, Wörter oder Sätze eingeben und dabei wird ein Passwort erstellt. Dies funktioniert darüber das die Eingabe in eine Liste gespeichert wird und dann durch die Funktion random.shuffle willkürlich verändert wird und dann wieder als String zurückgeliefert wird.

**Comment by dhbwJK23** [comment]
Sehr schön gemacht. Nützliche Anwendung so ein Passwort Generator!

**Comment by DGrillCoder** [comment]
Der Passwortgenerator ist eine gute Idee, vor allem mit den zwei Varianten.
Schere-Stein-Papier interessant gestaltet
